app/controllers/main/routes.py

```diff
@@ -49,50 +49,107 @@ def events(selectedTerm=None):
         currentTime = currentTime,
         user = g.current_user)
 
-@main_bp.route('/profile/<username>', methods = ['GET'])
-def profilePage(username):
-    if not g.current_user.isCeltsAdmin and not g.current_user.isCeltsStudentStaff and g.current_user.username != username:
-        return "Access Denied", 403
+@main_bp.route('/profile/<username>', methods=['GET'])
+def viewVolunteersProfile(username):
+    """
+    This function displays the information of a volunteer to the user
+    """
     try:
-        profileUser = User.get(User.username == username)
-        upcomingEvents = getUpcomingEventsForUser(username)
-        programs = Program.select()
-        interests = Interest.select().where(Interest.user == profileUser)
-        interests_ids = [interest.program.id for interest in interests]
-        rsvpedEventsList = EventRsvp.select().where(EventRsvp.user == profileUser)
-        rsvpedEvents = [event.event.id for event in rsvpedEventsList]
-
-        allUserEntries = list(BackgroundCheck.select().where(BackgroundCheck.user == profileUser))
-        completedBackgroundCheck = {entry.type.id: entry.passBackgroundCheck for entry in allUserEntries}
-        backgroundTypes = list(BackgroundCheckType.select())
-
-
-        return render_template('/volunteer/volunteerProfile.html',
-                               title="Volunteer Interest",
-                               user = profileUser,
-                               programs = programs,
-                               interests = interests,
-                               interests_ids = interests_ids,
-                               upcomingEvents = upcomingEvents,
-                               rsvpedEvents = rsvpedEvents,
-                               backgroundTypes = backgroundTypes,
-                               completedBackgroundCheck = completedBackgroundCheck
-                               )
+        volunteer = User.get(User.username == username)
+    except Exception as e:
+        return "User does not exist", 404
+    if (g.current_user == username) or g.current_user.isAdmin:
```

Comment:
You should use `volunteer` everywhere you can instead of `username`. Once you validate your inputs, use the validated data instead of the inputs. For this code it doesn't make much of a difference because you are ensuring `username` is valid above with the try/except, but it's a good principle to follow.

Reply:
We fixed the issue. Added **volunteer** everywhere we could.